File: tutorials/30-days-of-code/day-20-sorting/sorting.py
#!/bin/python3

import sys

n = int(input().strip())
a = list(map(int, input().strip().split(' ')))
# Write Your Code Here

swaps = 0

for num in range(n-1,0,-1):
  for i in range(num):
    if a[i]>a[i+1]:
      temp = a[i]
      a[i] = a[i+1]
      a[i+1] = temp
      swaps = swaps + 1

print('Array is sorted in ' + str(swaps) + ' swaps.')
print('First Element: ' + str(a[0]))
print('Last Element: ' + str(a[n-1]))

File: tutorials/30-days-of-code/day-20-sorting/sorting.js
function main() {
  var n = parseInt(readLine());
  a = readLine().split(' ');
  a = a.map(Number);
  // Write Your Code Here

  const firstElement = 0
  const lastElement = 0

  bubbleSort(a)

  console.log('Array is sorted in ' + numSwaps + ' swaps.')

  console.log('First Element: ' + firstElement)

  console.log('ast Element: ' + lastElement)

}

function bubbleSort(a) {
    for (let i = 0; i < n; i++) {
      // Track number of elements swapped during a single array traversal
      let numberOfSwaps = 0;

      for (let j = 0; j < n - 1; j++) {
        // Swap adjacent elements if they are in decreasing order
        if (a[j] > a[j + 1]) {
          swap(a[j], a[j + 1]);
          numberOfSwaps++;
        }
      }

      // If no elements were swapped during a traversal, array is sorted
      if (numberOfSwaps == 0) {
        break;
      }
    }
  }